modules/host.go

```diff
@@ -83,49 +93,113 @@ type (
+	// folders are used by the host to store the data that they contractually
+	// agree to manage.
+	StorageFolderManager interface {
+		// AddStorageFolder adds a storage folder to the host. The host may not
+		// check that there is enough space available on-disk to support as
+		// much storage as requested, though the host should gracefully handle
+		// running out of storage unexpectedly.
+		AddStorageFolder(path string, size uint64) error
+
+		// ForceRemoveStorageFolder removes a storage folder. The host will try
+		// to save the data on the storage folder by moving it to another
+		// storage folder, but if there are errors (such as not enough space)
+		// the host will forcibly remove the storage folder, discarding the
+		// data. This means that the host will be unable to provide storage
+		// proofs on the data, and is going to incur penalties.
+		ForceRemoveStorageFolder(index int) error
+
+		// RemoveStorageFolder will remove a storage folder from the host. All
+		// storage on the folder will be moved to other storage folders,
+		// meaning that no data will be lost. If the host is unable to save
+		// data, an error will be returned and the operation will be stopped.
+		RemoveStorageFolder(index int) error
+
+		// ResetStorageFolderHealth will reset the health statistics on a
+		// storage folder.
+		ResetStorageFolderHealth(index int) error
+
+		// ResizeStorageFolder will grow or shrink a storage folder in the
+		// host. The host may not check that there is enough space on-disk to
+		// support growing the storage folder, but should gracefully handle
+		// running out of space unexpectedly. When shrinking a storage folder,
+		// any data in the folder that needs to be moved will be placed into
+		// other storage folders, meaning that no data will be lost. If the
+		// host is unable to migrate the data, an error will be returned and
+		// the operation will be stopped.
+		ResizeStorageFolder(index int, newSize uint64) error
+
+		// StorageFolders will return a list of storage folders tracked by the
+		// host.
+		StorageFolders() []StorageFolderMetadata
+	}
+
 	// Host can take storage from disk and offer it to the network, managing things
 	// such as announcements, settings, and implementing all of the RPCs of the
 	// host protocol.
 	Host interface {
-		// Announce submits a host announcement to the blockchain, returning an
-		// error if its external IP address is unknown. After announcing, the
-		// host will begin accepting contracts.
-		Announce() error
+		// Announce submits a host announcement to the blockchain.  After
+		// announcing, the host will begin accepting contracts.
+		Announce(NetAddress) error
+
+		// ConfigureSmartPrices configures whether the host will automatically
+		// adjust prices based on the network environment. The algorithms aim
+		// to be profit maximizing.
+		ConfigureSmartPrices(contractPricing bool, downloadPricing bool, storagePricing bool, uploadPricing bool) error
```

Comment:
They aren't, and I've decided it's premature to be adding this function. It's the type of thing that should be pretty easy to tag on later.